runner/ollamarunner/runner.go

```diff
@@ -130,9 +130,9 @@ func (s *Server) NewSequence(prompt string, images []llm.ImageData, params NewSe
 	// Ensure that at least 1 input can be discarded during shift
 	params.numKeep = min(params.numKeep, s.cache.numCtx-1)
 
-	if int32(len(inputs)) > s.cache.numCtx {
-		discard := int32(len(inputs)) - s.cache.numCtx
+	discard := int32(len(inputs)) - s.cache.numCtx
```

Comment:
`discard` should be inside of the if statement - you should be able to just leave this logic as it is currently.